core/cmd/local_client.go

```diff
@@ -366,6 +366,44 @@ func (cli *Client) ResetDatabase(c *clipkg.Context) error {
 	return nil
 }
 
+func (cli *Client) HardResetDatabase(c *clipkg.Context) error {
+	logger.SetLogger(cli.Config.CreateProductionLogger())
+
+	prompt := NewTerminalPrompter()
+	var answer string
+	for {
+		answer = prompt.Prompt("Are you sure? This action is irreversible! (yes/No)")
+		if answer == "yes" {
+			break
+		} else if answer == "no" {
+			return nil
+		} else {
+			fmt.Printf("%s is not valid. Please type yes or no\n", answer)
+		}
+	}
+
+	if err := truncateDatabase(); err != nil {
+		logger.Errorw("failed to truncate database", "error", err)
+		return err
+	}
+	return nil
+}
+
+func truncateDatabase() error {
+	cfg := orm.NewConfig()
+	ormInstance, err := orm.NewORM(
+		cfg.DatabaseURL(),
+		cfg.DatabaseTimeout(),
+		gracefulpanic.NewSignal(),
+		cfg.GetDatabaseDialectConfiguredOrDefault(),
+		cfg.GetAdvisoryLockIDConfiguredOrDefault())
+	if err != nil {
+		return err
+	}
+
+	return ormInstance.TruncateDatabase()
```

Comment:
What about marking things as errored as we discussed?

Reply:
I looked into making BPTXM support errored/cancelled states, but couldn't figure out how to do it. I'm not confindent changing that code at all. I had a chat with @se3000 about it, and we decided it's not worth it at this point. Do you want to clarify Steve?

That being said, I'm happy to go over it again. I'll need your help identifying all the potential places where BPTXM needs to updated, thought.

Reply:
What I remember talking about was cutting the scope of sending replacement transactions for all of the nonces that were not yet confirmed. It was not clear to me that the behavior was always wanted, and it added a significant amount of scope to the story so I thought we'd be ok cutting the replacement transactions behavior. I don't recall discussing marking runs as cancelled or not, I thought the story was to nuke the database and clear records. Happy to jump on a call to clarify if it'd be quicker.